pkg/nodeagent/fornaxcore/grpc.go

```diff
@@ -0,0 +1,190 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fornaxcore
+
+import (
+	"context"
+	"io"
+	"time"
+
+	fornax "centaurusinfra.io/fornax-serverless/pkg/fornaxcore/grpc"
+	"centaurusinfra.io/fornax-serverless/pkg/nodeagent/retry"
+	"github.com/golang/protobuf/ptypes/empty"
+	grpc_retry "github.com/grpc-ecosystem/go-grpc-middleware/retry"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"k8s.io/klog/v2"
+)
+
+type FornaxCoreConfiguration struct {
+	endpoint       string
+	connTimeout    time.Duration
+	callTimeout    time.Duration
+	maxRecvMsgSize int
+}
+
+type FornaxCore interface {
```

Comment:
is this fornax core grpc service client? if it is, can we decorate its name to indicate so (e.g. FornaxCoreNode)?

Reply:
ack, changed it to FornaxCoreClient